### ISSUE TITLE
Info on deployed app should be ignored by git

### ISSUE THREAD

**Issue by sciurus** [aws/chalice#915]
When I run `chalice deploy`, the directory `.chalice/deployed/` is created and contain a json file that records the AWS resources for my deployment. Git is not configured to ignore this directory, when presumably it should be.

When I run `chalice new-project` a `.gitignore` file is created that contains `.chalice/deployments/`. I wonder if the value `deployments` there was meant to be `deployed` instead?

https://github.com/aws/chalice/blob/6a542f224f1dcc40b3237cedce02928bfe48d620/chalice/constants.py#L42

https://github.com/aws/chalice/blob/master/chalice/deploy/deployer.py#L831-L844

**Comment by cciotti** [comment]
If you [bootstrap the project](http://chalice.readthedocs.io/en/latest/quickstart.html#creating-your-project) using `chalice new-project` a [`.gitignore`](https://git-scm.com/docs/gitignore) is generated which will ignore that directory. 

```bash
$ chalice new-project foo
$ cat foo/.gitignore
.chalice/deployments/
.chalice/venv/
```

**Comment by sciurus** [comment]
I did bootstrap using `chalice new-project`. There are two similarly named directories:

* `.chalice/deployments/` contains zip files and is in gitignore
* `.chalice/deployed/` contains json files and is *not* in gitignore.

I'm happy to submit a PR adding the latter to [the gitignore template](https://github.com/aws/chalice/blob/6a542f224f1dcc40b3237cedce02928bfe48d620/chalice/constants.py#L41-L44) as well if there is agreement that it belongs there.

**Comment by jamesls** [comment]
Yeah, I think adding this to the default gitignore sounds reasonable.  People can always remove it if they do want to check in that file for some reason.

**Comment by skwashd** [comment]
@jamesls Ignoring `.chalice/deployments/` makes sense as it where the builds end up. 

Ignoring the `.chalice/deployed/` directory will result in each deployment creating a new API gateway endpoint. I don't think that's desired behaviour.

**Comment by surfkansas** [comment]
The `./chalice/deployed` folder is a critical component when using chalice across multiple developers that are targeting the same environment.  If this folder is .git ignored, each developer will deploy to a separate API Gateway endpoint.

Adding the .git ignore after project bootstrapping is relatively easy.  However, if we had the behavior changed, any developers using and depending on this folder would be in for a nasty surprise the next time they create a new project.

If this feature *is* implemented, I would recommend it as a --no-deployment-folder option to the new-project command, with the current behavior assumed if the switch isn't in place.

**Comment by AlexEshoo** [comment]
Does committing the deployed resourse manifest to VCS make any sense though? The deployed "state" is something that needs to always be up to date, the history of the state has no use when deploying. Imagine deploying based on that file while checked out to a different branch on the repository, there will still be errors. 

The way that state is handled with tools like terraform (for example) locates this file in a central location for all developers to poll against a "live" version of. See [the terraform docs](https://www.terraform.io/docs/state/index.html) on this for more details.
